packages/form-state-manager/src/utils/use-subscription.ts

```diff
@@ -54,16 +56,18 @@ export const initialMeta = (initial: any): Meta => ({
   visited: false
 });
 
-const useSubscription = ({ name, initialValue, clearOnUnmount, initializeOnMount, validate }: UseSubscription): SubscribtionData => {
+const useSubscription = ({ name, initialValue, clearOnUnmount, initializeOnMount, validate, subscription }: UseSubscription): SubscribtionData => {
   const { registerField, unregisterField, change, getFieldValue, blur, focus, formOptions } = useContext(FormManagerContext);
-  const [state, setState] = useState({
+  const [state, setState] = useState(() => ({
     value: initialValue,
     name,
-    meta: initialMeta(initialValue)
-  });
+    meta: initialMeta(initialValue),
+    internalId: generateId()
+  }));
   const {
     current: { getDetachedState, setDetachedState }
   } = useRef(createFieldState(state));
+  const [_, render] = useReducer((count) => count + 1, 0);
```

Comment:
Any specific reason why using reducer and not a state?

also `const [, render] => ....` should bee good enough. No need for the underscore.

Reply:
With `useState` you will have to still write `render((prev) => prev + 1))` so this seems to be a little bit cleaner in my eyes. Not sure what about performance, but Redux is using the same construction.